### PR TITLE
DBEX: adjust with_tracking of 526 Submission

### PR DESCRIPTION
**Pull request by freeheeling**
## Summary

- A not null violation error is raised within the `Form526JobStatusTracker` when a job status update is attempted for a nil submission ID
- The submission ID in the job tracker's "on error" handlers is populated by the `with_tracking` block, outside of which occur both `prepare_for_evss!` and `send_post_evss_notifications!` service calls
- Analyzing stack traces of raised errors indicated failures directly attributed to outage exceptions when calling EVSS's `/ratedDisabilities` endpoint in the `prepare_for_evss!` step
- Relocating calls to both `prepare_for_evss!` and `send_post_evss_notifications!` within the `with_tracking` block should address the nil submission ID error

Error handling has been refactored and conditional logic introduced to:
1. return early if an exception is raised when calling `prepare_for_evss!`
2. only send notifications if form submission is successful
3. prevent duplicate submission attempts in the event `send_post_evss_notifications!` raises an exception

## Related issue(s)

- [Datadog spikes](https://vagov.ddog-gov.com/logs?query=%22ActiveRecord%3A%3ANotNullViolation%22%20form526_submission_id%20&cols=host%2Cservice&index=%2A&messageDisplay=inline&refresh_mode=sliding&stream_sort=desc&viz=stream&from_ts=1704206104003&to_ts=1704810904003&live=true)
- [Sentry Error](https://vagov.ddog-gov.com/logs?query=%22ActiveRecord%3A%3ANotNullViolation%22%20form526_submission_id%20&cols=host%2Cservice&index=%2A&messageDisplay=inline&refresh_mode=sliding&stream_sort=desc&viz=stream&from_ts=1704206104003&to_ts=1704810904003&live=true)
- Zenhub ticket [#73266](https://app.zenhub.com/workspaces/disability-benefits-experience-team-carbs-6470c8bfffee9809b2634a52/issues/gh/department-of-veterans-affairs/va.gov-team/73266)

## Testing done

Corresponding spec runs successfully after refactoring code.

## What areas of the site does it impact?
Form 526 Submission notifications

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: app/models/concerns/form526_claim_fast_tracking_concern.rb [hunk 1]
```diff
@@ -76,7 +76,6 @@ def rrd_pdf_added_for_uploading?
   end
 
   def rrd_special_issue_set?
-    disabilities = form.dig('form526', 'form526', 'disabilities')
```
Comment (freeheeling) [inline app/models/concerns/form526_claim_fast_tracking_concern.rb:79]: Delete duplicate assignment already defined on L85 of original.

File: app/sidekiq/evss/disability_compensation_form/submit_form526.rb [hunk 2]
```diff
@@ -77,58 +77,61 @@ class SubmitForm526 < Job
...
 
         Sentry.set_tags(source: '526EZ-all-claims')
-        super(submission_id)
 
-        submission.prepare_for_evss!
+        # This instantiates the service as defined by the inheriting object
+        # TODO: this meaningless variable assignment is required for the specs to pass, which
+        # indicates a problematic coupling of implementation and test logic.  This should eventually
+        # be addressed to make this service and test more robust and readable.
+        service = service(submission.auth_headers)
```
Comment (freeheeling) [inline app/sidekiq/evss/disability_compensation_form/submit_form526.rb:89]: Had to relocate the service assignment to pass spec.

File: lib/sidekiq/form526_job_status_tracker/job_tracker.rb [hunk 3]
```diff
@@ -91,7 +91,8 @@ def with_tracking(job_title, saved_claim_id, submission_id, is_bdd = nil)
           exception_raised = true
           raise
         ensure
-          job_success unless exception_raised
+          job_status = Form526JobStatus.find_by(job_id: jid)
+          job_success unless exception_raised || job_status&.error_class
```
Comment (freeheeling) [inline lib/sidekiq/form526_job_status_tracker/job_tracker.rb:95]: This additional evaluation was needed to address adjustments made to `with_tracking` in `SubmitForm526#perform`.

File: app/sidekiq/evss/disability_compensation_form/submit_form526.rb [hunk 4]
```diff
@@ -77,58 +77,61 @@ class SubmitForm526 < Job
...
+              body = transform_service.transform(submission.form['form526'])
+              # 3. send transformed submission data to Lighthouse endpoint
+              service = BenefitsClaims::Service.new(icn)
+              raw_response = service.submit526(body)
+              # 4. convert response from Lighthouse to a FormSubmitResponse for further processing (claim_id, status)
+              raw_response_struct = OpenStruct.new({
+                                                     body: { claim_id: raw_response.body },
+                                                     status: raw_response.status
+                                                   })
+              response = EVSS::DisabilityCompensationForm::FormSubmitResponse
```
Comment (tommasina-va) [inline app/sidekiq/evss/disability_compensation_form/submit_form526.rb:121]: I'm curious if there's a reason you didn't set the conditional equal to `response` on L104?

**Review by tommasina-va — APPROVED** [review]
LGTM!

File: app/sidekiq/evss/disability_compensation_form/submit_form526.rb (hunk 4 above)
Comment (freeheeling) [reply app/sidekiq/evss/disability_compensation_form/submit_form526.rb:121]: In hindsight, I should have. I suppose I adopted what already existed without recognizing that improvement.

**Review by tommasina-va — APPROVED** [review]
Looks great! 🚢

**Pull request merged**